### PR TITLE
Email service

### LINKED ISSUE #103: Add Email Service
## Summary
If we want to use the email verification component and the let the user reset their password we are going to need a way to email them. This task is to create an email service on the backend that will handle sending emails.

## Technical Details
- Server
- Use sendgrid. You'll need an account and api key from them. The free plan is enough.
- Use the sendgrid/mail node package.

## Acceptance Criteria
- setup up sendgrid service and use it to send emails to real email addresses.

### PR DESCRIPTION
**Pull request by Linkaloo**
## Description
Provides endpoints for a user to state they forgot their password and so they can set a new password.
Provides mutations in the client to use and creates a simple page to apply a new password to the user.

## Related Issue
closes #103 

## Type of Changes
Please select the relevant option:
- [ ] Bug fix
- [x] New feature
- [x] Enhancement
- [ ] Documentation update
- [ ] Other (please specify)

## Things to Update
Please add any features, buttons, or pages that might need updating in the future:
- [ ] Button Functunality 
- [ ] Code Documentation 
- [x] Mobile Responsiveness 
- [x] Code testing 
- [ ] Other (please specify)

## Screenshots 
Email:
image

Reset password page:
image

## Additional Notes
I made a very simplistic password change page because I know Trevor is working on something. I wasn't sure what idea to go with this at first so just decided to run with this one:

Trevor made a reset password component, which is really a forgot password component as you ask for the email. That should be its own page that calls the forgot password endpoint. The verify email we will leave out for now and maybe instead use that for when a user adds their email. Then this will be the actual page where a password is changed and there is already a type of authentication done to ensure that it is the correct user.

### CODE REVIEW

File: client/src/store/api/api.slice.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@ import { getAccessToken } from '@/utils';
 
 export const apiSlice = createApi({
   baseQuery: fetchBaseQuery({
-    baseUrl: import.meta.env.VITE_API_BASE_URL,
+    // baseUrl: import.meta.env.VITE_API_BASE_URL,
+    baseUrl: 'http://localhost:3000/api',
```
Comment (KingJordan152) [inline client/src/store/api/api.slice.ts:7]: I think this should be changed back to `import.meta.env.VITE_API_BASE_URL`.

**Review by KingJordan152 — CHANGES_REQUESTED** [review]
This looks beyond amazing. Terrific work, Josh!

I left one comment regarding the `baseUrl` in the API slice. I believe that should be changed back to what you had previously. 

Also, don't forget to send us the send grid API key!

**Review by Aligary — APPROVED** [review]
Thanks, Josh 👍🏽 Everything looks well

**Review by dafneMena — APPROVED** [review]
This looks amazing! Love the layout and design, you nailed it!
I noticed you sent the grid API key to our emails so thank you for that as well.

**Review by KingJordan152 — APPROVED** [review]
Everything looks good. Thank you for implementing my request. Approved!

**Pull request merged**